src/layer/tether.js

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import React from 'react';
+import Popper from 'popper.js';
+import {toPopperPlacement, parsePopperOffset} from './utils.js';
+import {PLACEMENT} from './constants.js';
+import type {TetherPropsT, TetherStateT, PopperDataObjectT} from './types.js';
+
+class Tether extends React.Component<TetherPropsT, TetherStateT> {
+  static defaultProps = {
+    anchorRef: null,
+    ignoreBoundary: false,
```

Comment:
I often regret building this feature into the popover. It establishes a wrapper interface into the popper.js library for no substantial gain. we don't perform any manipulation on this value before provided to popper js. i think a more appropriate interface is to expose a `popperOptions` prop that get's passed to popper. This would enable devs to customize however they need to.

```js
this.popper = new Popper(this.props.anchorRef, this.props.popperRef, {
    // ... other parameters
    preventOverflow: {enabled: !this.props.ignoreBoundary},
    ...this.props.popoverOptions,
  },
});
```

Not sure if absolutely necessary for this pr, but seems like a good chance to introduce more productive patterns.

Reply:
relevant gh issue: https://github.com/uber-web/baseui/issues/1152